torch/distributed/device_mesh.py

```diff
@@ -553,7 +553,7 @@ def init_device_mesh(
                     f"Found len(mesh_dim_names): {len(mesh_dim_names)} and len(mesh_shape):{len(mesh_shape)}.",
                 )
 
-        mesh = torch.arange(math.prod(mesh_shape)).view(mesh_shape)
+        mesh = torch.arange(math.prod(mesh_shape), dtype=torch.int).view(mesh_shape)
```

Comment:
I think one additional place need to be change: https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L215

We need to change this `mesh.detach().cpu()` to sth like `mesh.detach().to(device="cpu", dtype=torch.int)